docker/pypi/wmagent/init.sh

```diff
@@ -296,18 +298,27 @@ check_databases() {
 }
 
 set_cronjob() {
-    stepMsg="Populating cronjob with utilitarian scripts for the $WMA_USER"
+    stepMsg="Populating cronjob with utilitarian scripts for user: $WMA_USER"
     echo "-----------------------------------------------------------------------"
     echo "Start: $stepMsg"
+    local errVal=0
 
-    chmod +x $WMA_DEPLOY_DIR/deploy/renew_proxy.sh $WMA_DEPLOY_DIR/deploy/restartComponent.sh
-
+    # Populating proxy related cronjobs
     crontab -u $WMA_USER - <<EOF
 55 */12 * * * $WMA_MANAGE_DIR/manage renew-proxy
 58 */12 * * * python $WMA_DEPLOY_DIR/deploy/checkProxy.py --proxy /data/certs/myproxy.pem --time 120 --send-mail True --mail alan.malta@cern.ch
 */15 * * * *  source $WMA_DEPLOY_DIR/deploy/restartComponent.sh > /dev/null
 EOF
+    let errVal+=$?
 
+    # Populating CouchDB related cronjobs
+    wmagent-couchapp-init
```

Comment:
Given that this script depends on the environment variable `WMAGENT_CONFIG`, won't it failed now that you removed it from the `manage` script?

I see it only in the `install.sh` script, which AFAICT, it is only executed when building the image.

Reply:
No, it won't fail.  The way how it was  set was a relic from the old manage. I tend to accumulate all environment variables at 3 places: 
* `Dockerfile` - everything already known at build time. Those follow the standard naming convention for environment variables -  `SNAKE_CASE_CAPITALIZED` Always prefixed with `WMA_`
* `bin/manage-common.sh` - everything that is auxiliary to the initialization process and is also needed from `bin/manage` , but not necessarily need to be environment variables. Two clean examples are:
   * variables from `WMAgent.secrets` - None of them must go in the environment, they better be contained only in the global scope of the currently executed script, so that they go away with it on exit. Those follow the standard naming convention we use for environment variables - `SNAKE_CASE_CAPITALIZED` 
   *  variables needed for pointing to the `.init*` flags - none of them are needed for the processes not involved in the initialization, so again no need to pollute the environment so they stay again only in the global scope of the executed  script (loaded  by sourcing the `bin/manage_common.sh` file itself). Those follow the standard convention - `camelCase` .
*  `$WMA_ENV_FILE` - all environment variables that are needed from `wmcoreD` or hardwired  somewhere in the WMCore code. Those are variable without which we cannot run the components. This file also serves as a single place for resolving any difference in naming of variables pointing to one and the same thing but having different names or following different naming conventions (if such) during the initialization process and the runtime code.

Said all that, the logical place for this variable is inside the `$WMA_ENV_FILE` , and definitely not exported locally at some random place inside `bin/manage` script. That's why I have it moved inside `$WMA_ENV_FILE`. Now it is always visible to everybody who is involved in the initialization process and also during the runtime. So no it won't break the code, it is in the correct place now. and I have tested it before making the commit.

Reply:
Todor, not to lose this information, can you please persist it in this documentation:
https://gitlab.cern.ch/dmwm/wmcore-docs/-/blob/master/docs/wmcore/WMAgent-in-Docker.md?ref_type=heads
?

Perhaps create a new sub-section named something like "Deployment and environment variables"

Reply:
Alan the information in this file is really really outdated 
I'll create the section, but just in case you have missed it, all of it is in the manage file itself as a set of comments, at  the file's header: https://github.com/dmwm/CMSKubernetes/blob/511dd27e1e3c4d2c95eefe4c9d0ed71ac6277828/docker/pypi/wmagent/bin/manage#L11-L58   . I added it there on request from Valentin.

Reply:
@todor-ivanov I don't think we should provide an extensive list of environment variables and what knot, as that will naturally become outdated at some point - like these, with in less than a couple of months are already outdated.

I think the overview of type of variables, their purpose and where they have been placed is what is really important and what the whole team will profit from.

Reply:
I updated the document already